src/modm/platform/uart/stm32/uart.hpp.in

```diff
@@ -75,28 +75,29 @@ public:
 	}
 
 	template< class SystemClock, baudrate_t baudrate, percent_t tolerance=pct(1) >
-	static void modm_always_inline
-%% if options["buffered"]
-	initialize(uint32_t interruptPriority = 12, Parity parity = Parity::Disabled)
-%% else
-	initialize(Parity parity = Parity::Disabled)
-%% endif
+	static inline void
+	initialize({% if buffered %}uint32_t interruptPriority, {% endif %}Parity parity, WordLength length)
```

Comment:
Can we please change the interface here to make code portable between buffered and unbuffered Uart?

```suggestion
	static inline void
	initialize(Parity parity, WordLength length{% if buffered %}, uint32_t interruptPriority, {% endif %})
```

Reply:
Yes, please. I was uncertain about breaking backwards compatibility.
I'm not at all in favor of bundling NVIC settings with the interfaces, mostly because of the issue with shared interrupts, which completely breaks the API promises (ie. effectively showing the same shared IRQ as different IRQs).

I would perhaps leave it like that for now, since it's backwards compatible and instead add a new driver call NVIC, that perhaps exposes IRQ handling in a better way. (perhaps with a connect-like API or something like that).

Reply:
Well… actually, we're already breaking the API with the lbuild option changes, so it makes sense to fix this now!

Reply:
If I were a user not closely monitoring modm development, I would generally prefer to have some changelogs telling me if and how my code gets broken. Could we get the xpcc changelog back without much effort, could we generate it semi-automatically? People are putting modm into real products, so we should care somehow ...

Reply:
Yes, I need to bring the changelog back… Ideally as part of the PR approval somehow, so that the mountain of work doesn't dull my motivation to do quarterly releases.
https://twitter.com/modm_embedded/status/1345396075813859328

Reply:
I'm bringing ~Sexy~ Changelog back
Them other libs don't know how to act